tests/components/dsmr_reader/test_config_flow.py

```diff
@@ -0,0 +1,70 @@
+"""Tests for the config flow."""
+from unittest.mock import MagicMock
+from homeassistant.components.dsmr_reader.const import DOMAIN
+from homeassistant.components.mqtt import DATA_MQTT
+from homeassistant.config_entries import SOURCE_IMPORT, SOURCE_USER
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import (
+    RESULT_TYPE_CREATE_ENTRY,
+    RESULT_TYPE_FORM,
+    RESULT_TYPE_ABORT,
+)
+
+from tests.common import MockConfigEntry
+
+
+async def test_import_step(hass: HomeAssistant):
+    """Test the import step."""
+    # configure bogus mqtt service to pass validation
+    hass.services.async_register("mqtt", "publish", None)
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_IMPORT},
+    )
+    assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == "DSMR Reader"
```

Comment:
Please add a test to make sure we can only import once

Reply:
Done. Thanks for catching this!